frontend/src/utilities/notebookControllerUtils.ts

```diff
@@ -577,6 +562,18 @@ export const useNotebookProgress = (
     }
   }
 
+  // If milestone steps are completed, mark off associated steps
+  Object.entries(AssociatedSteps).forEach(([key, values]) => {
+    if (progressSteps.find((p) => p.step === key)?.status === EventStatus.SUCCESS) {
+      values.forEach((value) => {
+        const currentStep = progressSteps.find((p) => p.step === value);
+        if (currentStep && currentStep.status !== EventStatus.SUCCESS) {
```

Comment:
```suggestion
        if (currentStep) {
```
Setting it to success doesn't matter if it's already there or not... (however if you were after something with this line, you can jump to the point with optional chaining)
```suggestion
        if (currentStep?.status !== EventStatus.SUCCESS) {
```